doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -629,12 +629,24 @@ file is a particular case of a module called a *library file*.
 
    Loads an OCaml plugin and its dependencies dynamically.  The :n:`@string`
    argument must be a valid `findlib <http://projects.camlcity.org/projects/findlib.html>`_
-   plugin name, for example ``coq-core.plugins.ltac``. The first component of
-   the plugin name is a package that has to be in scope of ``findlib``'s'
-   search path. As of Coq 8.16, the command also support a "legacy"
+   plugin name, for example ``coq-core.plugins.ltac``. As of Coq 8.16,
+   the command also support a "legacy"
    syntax compatible with the plugin loading system used in Coq
    8.0-8.15, see below.
 
+   The first component of the plugin name is a package that has to be in
+   scope of ``findlib``'s' search path. Conventionally, it is named
+   ``coq-pkg`` where ``pkg`` is the name of the plugin. It must
+   correspond to the name of the ``META.coq-pkg`` file.
+
```

Comment:
Findlib supports more ways to get such kind of metadata, in particular most packages actually install the a `META` file in `$lib/pkg/META` instead of in `$lib/META.pkg`. Moreover, it is not enough the META file exists, but it has to be in scope for findlib's search system to see it.

I'm not sure how much details to give about findlib here, because indeed it is going to be hard to cover all possible ways of generating and accessing findlib metadata. 

I'd rather be more synthetic here and just state that that package needs to be able to be found in `ocamlfind list`, and put a reference to the concrete plugin tutorials for concrete approaches at generating the metadata.

Reply:
Thanks @ejgallego !
I am not aware of all the possible ways. I tried to add the explanations I missed when porting a plugin from 8.15 to 8.16. Maybe various examples can be given?

Reply:
Yes, I think it is a good idea to add some more information there, I guess the main question is if we want to add it there, of some part of it should go into the coq-makefile documentation and we link it from there?

Maybe we can add a subsection in https://coq.github.io/doc/master/refman/practical-tools/utilities.html#building-a-coq-project-with-coq-makefile-details named "Setting up a plugin", and link from there?

So we could write something like: "If you are writing a plugin, you need to generate the right metadata so findlib can locate your plugin. This usually involves generating some kind of "META" file and placing it in a place where findlib can see it. Different build systems provide different helpers to do this, for coq_makefile, see here (link), for Dune see here, "

Reply:
It seems good to me. As you suggest, I think that it is important to link this part of the documentation from the documentation of `Declare ML Module` anyway, because it is the command that fails when something's wrong.

Reply:
About adding a section "Setting up a plugin", note that we have the currently almost empty [Writing Coq libraries and plugins](https://coq.inria.fr/refman/using/libraries/writing.html). FTR I would like to eventually get the relevant commands such as `Declare ML Module` moved there.